python/python/lance/torch/kmeans.py

```diff
@@ -199,7 +200,11 @@ def _count_rows_in_clusters(part_ids: List[torch.Tensor], k: int) -> torch.Tenso
         return num_rows
 
     def _fit_once(
-        self, data: torch.utils.data.IterableDataset, epoch: int, last_dist: float = 0.0
+        self,
+        data: torch.utils.data.IterableDataset,
+        epoch: int,
+        last_dist: float = 0.0,
+        column: Optional[str] = None,
```

Comment:
do we need column here? can we make sure that the dataset already yield the correct shape?

Reply:
Yes - this is for the pq codes. I had significant performance issues during pq training before adding this (and making a few other changes). It may not be the only solution, but it seems like the least intrusive to me.